internal/storage/fs/azblob/azblob_fs.go

```diff
@@ -0,0 +1,134 @@
+package azblob
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"time"
+
+	"go.flipt.io/flipt/internal/storage/fs/blob"
+	"go.uber.org/zap"
+
+	gcblob "gocloud.dev/blob"
+	_ "gocloud.dev/blob/azureblob"
+)
+
+// FS is only for accessing files in a single bucket. The directory
+// entries are cached. It is specifically intended for use by a source
+// that calls fs.WalkDir and does not fully implement all fs operations
+type FS struct {
+	logger *zap.Logger
+
+	// configuration
+	containerName string
+	urlstr        string
+
+	// cached entries
+	dirEntry *blob.Dir
+}
+
+// ensure FS implements fs.FS aka Open
+var _ fs.FS = &FS{}
+
+// ensure FS implements fs.StatFS aka Stat
+var _ fs.StatFS = &FS{}
+
+// ensure FS implements fs.ReadDirFS aka ReadDir
+var _ fs.ReadDirFS = &FS{}
+
+// New creates a FS for the container
+func NewFS(logger *zap.Logger, schema string, containerName string) (*FS, error) {
+	return &FS{
+		logger:        logger,
+		containerName: containerName,
+		urlstr:        fmt.Sprintf("%s://%s", schema, containerName),
+	}, nil
+}
+
+// Open implements fs.FS. it fetches the object contents from azure blob.
+func (f *FS) Open(name string) (fs.File, error) {
+	if name == "." {
+		return f.dirEntry, nil
+	}
+	pathError := &fs.PathError{
+		Op:   "Open",
+		Path: name,
+		Err:  fs.ErrNotExist,
+	}
+	if !fs.ValidPath(name) {
+		pathError.Err = fs.ErrInvalid
+		return nil, pathError
+	}
+
+	ctx := context.TODO()
```

Comment:
```suggestion
	ctx := context.Background()
```

I think we should use `context.Background()` throughout this file instead of `context.TODO()`